homeassistant/components/unifiprotect/sensor.py

```diff
@@ -782,3 +828,72 @@ def _async_update_device_from_protect(self, device: ProtectDeviceType) -> None:
         self._set_event_attrs(event)
         if event.end:
             self._async_event_with_immediate_end()
+
+
+class ProtectPTZSensor(ProtectDeviceEntity, SensorEntity):
+    """A sensor for Unifi PTZ Cameras.
+
+    It exposes the current preset as the state value and pan/tilt/zoom as state attributes.
+    """
+
+    device: Camera
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        data: ProtectData,
+        device: ProtectDeviceType,
+        description: EntityDescription,
+    ) -> None:
+        """Override the constructor to include the device name."""
+        super().__init__(data, device, description)
+
+        self._hass = hass
+
+        self._attr_unique_id = f"{device.name}_{description.key}"
```

Comment:
Names are not allowed in the unique id
https://developers.home-assistant.io/docs/entity_registry_index?_highlight=unique&_highlight=id#unique-id-requirements

Ideally we don't need to override `__init__` here

Reply:
Without this it was only using the description key, with this it seemed to follow the naming scheme of other entities on this device. I wasn't able to untangle how that was supposed to happen, will have to set aside some more time to dig further.